test/e2e/conformance/crd_partial_metadata_test.go

```diff
@@ -143,3 +150,232 @@ func TestPartialMetadataCRD(t *testing.T) {
 		require.True(t, enabled)
 	}
 }
+
+func TestPartialMetadataSameCRDMultipleWorkspaces(t *testing.T) {
```

Comment:
This test exercises the partialmetadata issue. 

Do you think we should also cover the watch-cache issue with a new test?

Reply:
We can probably add to the existing watch cache test?